### ISSUE TITLE
A way to get large number of activations

### ISSUE THREAD

**Issue by jmortizs** [NVIDIA/DIGITS#951]
I need classify more than five thousand images with a model I trained and I need to get the activations of the the first convolutional layer for each image, with the classify one option I can get those activations but it is not viable do that "manually" so I wonder if there is a way to do that "automatic", I thought of a script but don't know how.
Thanks

**Comment by lukeyeager** [comment]
I can't think of an easy way to do that, no. Not without editing/writing a fair amount of code.

One thing you could do to speed up each individual request would be to only generate activations for the layer that you care about by editing this line:
https://github.com/NVIDIA/DIGITS/blob/v4.0.0/digits/model/images/classification/views.py#L342

@gheinrich: no way to use `tools/inference.py` easily, right? That's still going to take a lot of scripting work.

**Comment by gheinrich** [comment]
Suppose the activations you're interested in are the output of a layer that has a `top` named `convX`. You could make that `top` an output of the network only in the `deploy` network:
```
layer {
  # Identity layer
  name: "convXoutput"
  bottom: "convX"
  top: "convXoutput"
  type: "Power"
  power_param { scale: 1.0 }
  include { stage: "deploy" }
}
```

Then do a classify many from command line and retrieve `convXoutput` in the output JSON: 
```
curl localhost:5000/models/images/classification/classify_many.json -XPOST -F job_id=20151129-1613289ffb -F image_list=@/path/to/image-list.txt
```

Disclaimer: it won't work out of the box because DIGITS expects the output of a classification network to be just a `SoftMax`. But if you're willing to go through one or two additional steps to convert your model into a generic one then you should be alright and you wouldn't have to change the code (not too much at least).

**Comment by gheinrich** [comment]
@jmortizs any feedback please?

**Comment by jmortizs** [comment]
@gheinrich I manage to do it with torch and python outside digits, did not try to much what you suggested because I was in a hurry.
